### PR TITLE
Use nightly instead of canary on Windows

### PR DESCRIPTION
**Pull request by simonhong**
Test - ProductInstallDetailsTest and MakeProductDetailsTest

Issue: https://github.com/brave/brave-browser/issues/394

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [ ] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.

## Test Plan:
`yarn test brave_unit_tests --filter=*Product*`

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions

### CODE REVIEW

File: chromium_src/chrome/install_static/chromium_install_modes.cc [hunk 1]
```diff
@@ -124,21 +124,21 @@ const InstallConstants kInstallModes[] = {
         sizeof(kInstallModes[0]),
         CANARY_INDEX,  // The mode for the side-by-side canary channel.
```
Comment (darkdh) [inline chromium_src/chrome/install_static/chromium_install_modes.cc]: also change this to `NIGHTLY_INDEX` along with the header and tests?

File: chromium_src/chrome/install_static/brave_product_install_details_unittest.cc [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        L"\\Application\\brave.exe",
+        DEVELOPER_INDEX, true, L"",
+    },
+    {
+        L"C:\\Users\\user\\AppData\\Local\\BraveSoftware\\Brave-Browser-Development"
+        L"Application\\brave.exe",
+        DEVELOPER_INDEX, false, L"",
+    },
+};
+#endif  // !GOOGLE_CHROME_BUILD
```
Comment (darkdh) [inline chromium_src/chrome/install_static/brave_product_install_details_unittest.cc]: should be `!OFFICIAL_BUILD`

File: chromium_src/chrome/install_static/chromium_install_modes.cc (hunk 1 above)
Comment (simonhong) [reply chromium_src/chrome/install_static/chromium_install_modes.cc]: Done.

File: chromium_src/chrome/install_static/brave_product_install_details_unittest.cc (hunk 2 above)
Comment (simonhong) [reply chromium_src/chrome/install_static/brave_product_install_details_unittest.cc]: Done.

**Comment by simonhong** [comment]
@darkdh Thanks for review! 😄

**Pull request merged**